barretenberg/cpp/src/barretenberg/crypto/merkle_tree/indexed_tree/content_addressed_indexed_tree.hpp

```diff
@@ -284,10 +284,7 @@ ContentAddressedIndexedTree<Store, HashingPolicy>::ContentAddressedIndexedTree(s
     zero_hashes_[0] = current;
 
     TreeMeta meta;
-    {
-        ReadTransactionPtr tx = store_->create_read_transaction();
-        store_->get_meta(meta, *tx, false);
-    }
+    store_->get_meta(meta);
```

Comment:
You will see this in a few places. I got rid of the need to pass a transaction just to read uncommitted meta data. The transaction was unused.